### DISCUSSION TITLE
Getting IP addresses of interfaces

### DISCUSSION THREAD

**Discussion by domrachev03** [ANLAB-KAIST/KENSv3#90] [Project Q&A 3. Internet Layer (IP/Routing)]
I am a bit confused about how should I get the IP address of all the interfaces? Should we create a sockets for each interface our device have? Or there is another way how could we get them?

**Comment by Pusnow** [comment]
You can use `getIPAddr()` method. See https://github.com/ANLAB-KAIST/KENSv3/wiki/Tip:-Figuring-out-the-source-IP-address-to-use#retrieve-the-source-ip-address-from-the-nic-port .
